File: src/DemoBattle/StraightFsApi/VillainController.fs
namespace StraightFsApi

open Common
open Types
open Newtonsoft.Json
open System.IO
open System.Web.Http

type VillainController() =
    inherit ApiController()

    member this.Get () =
        JsonConvert.DeserializeObject<FighterRepresentation[]>(File.ReadAllText(CommonStrings.VillainStoragePath))

File: src/DemoBattle/StraightCsApi/VillainController.cs
using System.Collections.Generic;
using System.Web.Http;
using System.Web.Http.Cors;
using System.IO;
using Common;
using Newtonsoft.Json;

namespace StraightCsApi
{
    [Route("api/villain")]
    [EnableCors(origins: "*", headers: "*", methods: "*")]
    public class VillainController : ApiController
    {
        public IEnumerable<Fighter> Get() =>
            JsonConvert.DeserializeObject<Fighter[]>(File.ReadAllText(CommonStrings.VillainStoragePath));
    }
}
